clang/lib/Tooling/DependencyScanning/DependencyScanningTool.cpp

```diff
@@ -231,6 +243,39 @@ class IncludeTreePPConsumer : public PPIncludeActionsConsumer {
 };
 } // namespace
 
+void IncludeTreePPConsumer::initialize(CompilerInstance &CI) {
+  if (llvm::Error E = PrefixMapping.configurePrefixMapper(
+          CI.getInvocation(), PrefixMapper)) {
+    ErrorToReport = std::move(E);
+    return;
+  }
+
+  auto ensurePathRemapping = [&]() {
```

Comment:
What is this lambda used for?

Reply:
it's called right below it. I found it convenient for using early exits.

Reply:
Am I missing something? Couldn't you could early-exit without the lambda since there's no code after it?

Reply:
But that would prevent adding more code in the future; you either have to notice and avoid adding more code afterwards or change the code and remove the `return`s.
This way the code is self-contained and doesn't impose a restriction on how you expand this function.